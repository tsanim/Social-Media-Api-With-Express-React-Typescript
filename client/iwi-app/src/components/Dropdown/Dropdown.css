.dropbtn {
    color: white;
    padding: 16px;
    font-size: 16px;
}

.dropdown {
    position: relative;
    display: inline-block;
}

.dropdown-content {
    display: none;
    position: absolute;
    background-color: #8AAAE5;
    min-width: 160px;
    box-shadow: 0px 8px 16px 0px rgba(0, 0, 0, 0.2);
    z-index: 1;
}

.dropdown-content a {
    color: black;
    padding: 12px 16px;
    text-decoration: none;
    display: block;
}

.dropdown:hover .dropdown-content {
    display: block;
}

.dropdown:hover .dropbtn {
    color: rgb(65, 65, 124);
}