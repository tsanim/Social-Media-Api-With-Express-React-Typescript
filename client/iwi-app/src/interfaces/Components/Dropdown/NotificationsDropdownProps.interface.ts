import Notification from "../../Feed/Notification.interface";

export default interface NotificationsDropdownProps {
    notifications: Notification[];
}