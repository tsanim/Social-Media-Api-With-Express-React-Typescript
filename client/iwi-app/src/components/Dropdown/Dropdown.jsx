import React from 'react';
import { Link } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import './Dropdown.css';
import { faBell } from '@fortawesome/free-solid-svg-icons';
import PropTypes from 'prop-types';

function Dropdown(props) {
    return (
        <div className="dropdown">
            <button className="dropbtn"><FontAwesomeIcon icon={faBell} />{props.notifications.length}</button>
            <div className="dropdown-content">
                {
                    props.notifications.map(n => <Link key={n._id} to={`/chat?sender=${n.sender}`}> {n.message} </Link>)
                }
            </div>
        </div>
    )
}

Dropdown.propTypes = {
    notifications: PropTypes.array,
}

export default Dropdown;