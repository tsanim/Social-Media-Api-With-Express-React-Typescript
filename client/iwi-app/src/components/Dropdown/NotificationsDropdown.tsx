import React from 'react';
import { Link } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import './Dropdown.css';
import { faBell } from '@fortawesome/free-solid-svg-icons';
import PropTypes from 'prop-types';
import DropdownProps from '../../interfaces/Components/Dropdown/NotificationsDropdownProps.interface';

function Dropdown(props: DropdownProps) {
    return (
        <div className="dropdown">
            <button className="dropbtn"><FontAwesomeIcon icon={faBell} />{props.notifications.length}</button>
            <div className="dropdown-content">
                {
                    props.notifications.map(n => <Link key={n._id} to={`/chat?sender=${n.sender}`}> {n.message} </Link>)
                }
            </div>
        </div>
    )
}

Dropdown.propTypes = {
    notifications: PropTypes.array,
}

export default Dropdown;