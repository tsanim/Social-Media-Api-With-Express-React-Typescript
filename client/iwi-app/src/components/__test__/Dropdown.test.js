import React from 'react';
import Dropdown from '../Dropdown/Dropdown';

import { shallow } from 'enzyme';
import { Link } from 'react-router-dom';

function dropdownComponentSetup() {
    const props = {
        notifications: [ { _id: '123', sender: '' } ],
    }


    const enzymeWrapper = shallow(<Dropdown {...props} />)

    return {
        props,
        enzymeWrapper
    }
}

describe('Dropdown component', () => {
    it('should render initial layout', () => {
        const { enzymeWrapper } = dropdownComponentSetup();

        expect(enzymeWrapper.getElements()).toMatchSnapshot();
    });

    it('should render div container with class "dropdown"', () => {
        const { enzymeWrapper } = dropdownComponentSetup();

        expect(enzymeWrapper.find('div.dropdown')).toHaveLength(1);
    });

    it('should render button tag with class "dropbtn"', () => {
        const { enzymeWrapper } = dropdownComponentSetup();

        expect(enzymeWrapper.find('button.dropbtn')).toHaveLength(1);
    });

    it('should render div tag with class "dropdown-content"', () => {
        const { enzymeWrapper } = dropdownComponentSetup();

        expect(enzymeWrapper.find('div.dropdown-content')).toHaveLength(1);
    });

    it('should render div tag children with count equal to notifications array length', () => {
        const { enzymeWrapper, props } = dropdownComponentSetup();

        expect(enzymeWrapper.find('div.dropdown-content').children()).toHaveLength(props.notifications.length);
    });

    it('should render Link Components count equal to notifications array length', () => {
        const { enzymeWrapper, props } = dropdownComponentSetup();

        expect(enzymeWrapper.find(Link)).toHaveLength(props.notifications.length);
    });
});