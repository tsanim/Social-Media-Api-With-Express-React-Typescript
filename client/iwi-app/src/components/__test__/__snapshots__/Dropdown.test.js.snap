// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown component should render initial layout 1`] = `
Array [
  <div
    className="dropdown"
  >
    <button
      className="dropbtn"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              448,
              512,
              Array [],
              "f0f3",
              "M224 512c35.32 0 63.97-28.65 63.97-64H160.03c0 35.35 28.65 64 63.97 64zm215.39-149.71c-19.32-20.76-55.47-51.99-55.47-154.29 0-77.7-54.48-139.9-127.94-155.16V32c0-17.67-14.32-32-31.98-32s-31.98 14.33-31.98 32v20.84C118.56 68.1 64.08 130.3 64.08 208c0 102.3-36.15 133.53-55.47 154.29-6 6.45-8.66 14.16-8.61 21.71.11 16.4 12.98 32 32.1 32h383.8c19.12 0 32-15.6 32.1-32 .05-7.55-2.61-15.27-8.61-21.71z",
            ],
            "iconName": "bell",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        title=""
        transform={null}
      />
      1
    </button>
    <div
      className="dropdown-content"
    >
      <Link
        to="/chat?sender="
      >
         
         
      </Link>
    </div>
  </div>,
]
`;
